test/v1_1_0/upgrade/UpgradeFrom_v1_1_0.t.sol

```diff
@@ -0,0 +1,345 @@
+        {
+            GaugeVote[] memory vote = new GaugeVote[](1);
+            vote[0] = GaugeVote(1, gauge);
+            bobVPSnapshot = escrow.votingPower(bobToken);
+            voter.vote(bobToken, vote);
+        }
+        vm.stopPrank();
+
+        // carol begins unstaking
+        vm.startPrank(CAROL_ADDRESS);
+        {
+            lock.approve(address(escrow), carolToken);
+            escrow.beginWithdrawal(carolToken);
+        }
+        vm.stopPrank();
+
+        // wait a bit
+        vm.warp(4 weeks);
+
+        vm.startPrank(DAVID_ADDRESS);
+        {
+            lock.approve(address(escrow), davidToken);
+            escrow.beginWithdrawal(davidToken);
+        }
+        vm.stopPrank();
+    }
+
+    function testValidateUpgradeGaugeVoter__v1_0_0__v1_1_0() public {
+        Options memory options;
+
+        string[] memory exclude = new string[](1);
+        // disable initializers is invoked but the custom unsafe allow option is not set in the natspec
+        exclude[0] = "lib/osx/packages/contracts/src/core/plugin/PluginUUPSUpgradeable.sol";
+        options.exclude = exclude;
+
+        options.referenceContract = "SimpleGaugeVoter.sol";
+        Upgrades.validateUpgrade("SimpleGaugeVoter_v1_1_0.sol:SimpleGaugeVoterV1_1_0", options);
+    }
+
+    function testValidateUpgradeLock__v1_0_0__v1_1_0() public {
+        Options memory options;
+
+        options.referenceContract = "Lock.sol";
+        Upgrades.validateUpgrade("Lock_v1_1_0.sol:LockV1_1_0", options);
+    }
+
+    function testInitialState() public view {
+        // alice is locked and has voting power
+        assertEq(escrow.locked(aliceToken).amount, 1_000 ether);
+        assertGt(escrow.votingPower(aliceToken), 1_000 ether);
+
+        // bob is locked and is currently voting
+        assertEq(escrow.locked(bobToken).amount, 1_000 ether);
+        assertTrue(voter.isVoting(bobToken));
+        assertEq(voter.votes(bobToken, gauge), bobVPSnapshot);
+
+        // carol is locked and can exit
+        assertEq(escrow.locked(carolToken).amount, 1_000 ether);
+        assertTrue(queue.canExit(carolToken));
+
+        // david is locked and cannot exit
+        assertEq(escrow.locked(davidToken).amount, 1_000 ether);
+        assertFalse(queue.canExit(davidToken));
+        assertEq(queue.ticketHolder(davidToken), DAVID_ADDRESS);
+    }
+
+    function testUpgrade() public {
+        // simple upgrade for testing
+        // deploy the new implementations
+        // LockV1_1_0 lockV1_1_0 = new LockV1_1_0();
+        // Lock lockV1_1_0 = new Lock();
+        // SimpleGaugeVoterV1_1_0 voterV1_1_0 = new SimpleGaugeVoterV1_1_0();
+
+        // upgrade the contracts
+        vm.startPrank(address(dao));
+        {
+            // unsafe upgrade
+            // lock.upgradeTo(address(lockV1_1_0));
+            // voter.upgradeTo(address(voterV1_1_0));
+
+            // safe upgrade
+            _safeUpgradeVoter(address(voter));
```

Comment:
I'd still use `upgradeTo` instead of `upgradeProxy` helper. The reason being is in the actual upgrade, we don't use `upgradeProxy`. To keep things as close as possible, both the tests and the actual upgrade must follow the same flow.

Storage compatibility can still be tested without `upgradeProxy`.

Reply:
that's a fair point. Ideally we'd use a single upgrade factory

Reply:
Javi working on this for resets